pkg/ipcache/metadata.go

```diff
@@ -67,6 +67,14 @@ type metadata struct {
 	// m is the actual map containing the mappings.
 	m map[netip.Prefix]prefixInfo
 
+	// overriddenIdentities tracks which prefixes have an override identity
+	// assigned to them and therefore do not need its identity released when
+	// updated or removed.
+	// overriddenIdentitiesMu protects overriddenIdentities and must be taken
+	// both before the metdata.RWMutex or the IPCache.Lock
+	overriddenIdentitiesMu lock.Mutex
+	overriddenIdentities   map[netip.Prefix]struct{}
```

Comment:
I am unfortunately not super familiar with identity allocation. I chose the separate map because it more closely mirrors the existing behavior. If I understand your implementation correctly, it's main idea is to determine the identity based on the labels instead of the numeric identity, correct?

My main worry here was that for CiliumNode objects (which can get assigned an override identity via CEW), we wouldn't have the labels available that lead to the identity being assigned to the node. But re-reading the code, it seems that the labels from the CEW are written into the CiliumNode object together with the assigned identity, so that should work too.

The other open question I have is around the policy map update. Here I'm out of my depth if it makes sense to update the policy maps for "override identities", which I assume are already added to the policy map via policy and if this is something we need to track here, or if it is something that is anyway already tracked by the selector cache.

Reply:
In order to keep the code simpler, I decided to not treat override identities special when it comes to notifying the policy engine and have InjectLabels do the notification for all identities. Let's see how that works out.